src/App.tsx

```diff
@@ -1,26 +1,57 @@
-import React from 'react';
-import logo from './logo.svg';
+import React, { useEffect, useState } from 'react';
+import TopElements from "./components/topElements";
+import TotalPrice from "./components/totalPrice";
+import ShoppingList from "./components/shoppingList";
+import { IShopping } from "./utils/interfaces";
 import './App.css';
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.tsx</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+const bought: IShopping[] = [
+    {
+        place: 'shop1',
+        price: 10000,
+        date: '03.04.20',
+        isEdit: false,
+    },
+    {
+        place: 'shop2',
+        price: 12000,
+        date: '05.11.20',
+        isEdit: false,
+    },
+    {
+        place: 'shop3',
+        price: 8000,
+        date: '25.07.21',
+        isEdit: false,
+    },
+]
+
+const App = (): JSX.Element => {
+    const [purchase, setPurchase] = useState<IShopping>()
+    const [shoppingList, setShoppingList] = useState<IShopping[]>(bought)
+
+    useEffect(() => {
+        if (purchase) {
+            setShoppingList(() => [...shoppingList, purchase])
+        }
+    }, [purchase])
+
+    console.log(shoppingList)
+
+    const getPurchase = (purchase: IShopping) => {
+        const {place, price, date} = purchase;
+        setPurchase({place, price, date, isEdit: false})
+    }
```

Comment:
Why does your function have a name getPurchase and inside of it you call setPurchase? It's supposed to either be get or set.